### ISSUE TITLE
False positive require-store-reactive-access with melt-ui

### ISSUE THREAD

**Issue by mrkvon** [sveltejs/eslint-plugin-svelte#1312]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.12.0

### What did you do?

I initialized a svelte-kit project

I have a [menu](https://www.melt-ui.com/docs/builders/dropdown-menu) built with @melt-ui/svelte v0.86.6

```svelte
<script lang="ts">
	import { createDropdownMenu } from '@melt-ui/svelte';

	const {
		elements: { menu, item, trigger }
	} = createDropdownMenu();
</script>

<button {...$trigger} use:trigger>open menu</button>

<div {...$menu} use:menu>
	<div {...$item} use:item>first item</div>
	<div {...$item} use:item>second item</div>
</div>
```

I run `npx eslint .`, and receive the errors.

### What did you expect to happen?

No error reported.

### What actually happened?

```
   9:27  error  Use the $ prefix or the get function to access reactive values instead of accessing the raw store  svelte/require-store-reactive-access
  11:21  error  Use the $ prefix or the get function to access reactive values instead of accessing the raw store  svelte/require-store-reactive-access
  12:22  error  Use the $ prefix or the get function to access reactive values instead of accessing the raw store  svelte/require-store-reactive-access
  13:22  error  Use the $ prefix or the get function to access reactive values instead of accessing the raw store  svelte/require-store-reactive-access
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/mrkvon/svelte-lint-error-reproduction

### Additional comments

Seems to be related to the dual nature (store & action) of variables generated by melt-ui.

**Comment by baseballyama** [comment]
According to the Melt UI documentation, it says to write it like <button use:melt={$trigger}>, but is your code really correct?


https://www.melt-ui.com/docs/builders/dropdown-menu

```svelte
<script lang="ts">
  import { createDropdownMenu, melt } from '@melt-ui/svelte'
  const {
    elements: { menu, item, trigger, arrow }
  } = createDropdownMenu()
</script>
 
<button use:melt={$trigger}>Click me</button>
<div use:melt={$menu}>
  <div use:melt={$item}>...</div>
  <div use:melt={$item}>...</div>
  <div use:melt={$item}>...</div>
  <div use:melt={$arrow} />
</div>
```

**Comment by mrkvon** [comment]
Yes, I'm pretty confident it is a correct code without [melt-ui preprocessor](https://www.melt-ui.com/docs/preprocessor). There is a switch in the documentation (I see it at bottom left):

<img width="179" height="49" alt="Image" src="https://github.com/user-attachments/assets/7ae74713-0643-44b3-87c2-928313cd14ef" />

and if you switch it off, the same section shows:

```ts
<script lang="ts">
  import { createDropdownMenu, melt } from '@melt-ui/svelte'
  const {
    elements: { menu, item, trigger, arrow }
  } = createDropdownMenu()
</script>
 
<button {...$trigger} use:trigger>Click me</button>
<div {...$menu} use:menu>
  <div {...$item} use:item>...</div>
  <div {...$item} use:item>...</div>
  <div {...$item} use:item>...</div>
  <div {...$arrow} use:arrow />
</div>
```

I can check whether the syntax with preprocessor suppresses the false positive. In either case, this issue remains valid.

**Comment by mrkvon** [comment]
I can confirm that using the melt-ui preprocessor suppresses the error. So that syntax can be used until the underlying issue is resolved.

**Comment by baseballyama** [comment]
Thank you for the information.
Basically, Actions (the `foo` part in `use:foo`) must be function. So when I see `<div {...$item} use:item>`, it seems like the value of the `item` store would need to be both an `object` and a `function` at the same time, which sounds impossible.
I understand that the Melt UI preprocessor transforms this code into valid code, but since `eslint-plugin-svelte` runs linting against the preprocessed code, I don’t think it’s possible to lint it correctly.
From the perspective of ecosystem compatibility, I recommend writing code without relying on the preprocessor.

Anyway, there’s nothing that `eslint-plugin-svelte` can do, so I’ll close this issue.

**Comment by mrkvon** [comment]
The issue specifically concerns the situation when **preprocessor is not used at all**. Melt-ui indeed produces things that are both store and action. It is possible and valid for something in TypeScript to be both object and function, and `eslint-plugin-svelte` handles this situation incorrectly by producing false positive.

This is a valid issue and ***should remain open***, even if it may be difficult to address.

The triggered rule seems to incorrectly assume that if something is a store, it can't be anything else.

**Comment by baseballyama** [comment]
Yes, that’s true. It is indeed possible to be both a function and an object.
I think this could be resolved if type information were available.
